### PR TITLE
Save all edited scenes when Runing (and auto-save is enabled).

### LINKED ISSUE #4410: Play the project is only saving the current selected tab
Play the project is only saving the current selected tab.

**Steps to reproduce:**
1. Create two scenes(tabs)
2.Edit both to that the tabs show *
3.Press "Play the project" -> BUG: Now only the selected tab is beeing saved, the other tab still shows * and the game is not showing the changes of the second(not saved) tab.

### PR DESCRIPTION
**Pull request by MarianoGnu**
Also add a new shortcut to save all modified scenes previously saved. (Ctrl+Alt+Shift+S)

### CODE REVIEW

File: tools/editor/editor_data.cpp [hunk 1]
```diff
@@ -611,11 +611,16 @@ void EditorData::set_edited_scene(int p_idx){
 	current_edited_scene=p_idx;
 	//swap
 }
-Node* EditorData::get_edited_scene_root(){
+Node* EditorData::get_edited_scene_root(int idx){
 
 	ERR_FAIL_INDEX_V(current_edited_scene,edited_scene.size(),NULL);
```
Comment (akien-mga) [inline tools/editor/editor_data.cpp]: I get this should be in the first `if` block (same for the other methods with an added index).

Comment (MarianoGnu) [inline tools/editor/editor_data.cpp]: Sorry, i didn't get the point, should i move the error check inside the if block?

Comment (akien-mga) [inline tools/editor/editor_data.cpp]: Yeah I mean checking for the validity of `current_edited_scene` is only relevant when you're going to use it, so the code should be:
```
	if (idx < 0) {
		ERR_FAIL_INDEX_V(current_edited_scene,edited_scene.size(),NULL);
		return edited_scene[current_edited_scene].root;
	} else {
		ERR_FAIL_INDEX_V(idx,edited_scene.size(),NULL);
		return edited_scene[idx].root;
	}

**Comment by reduz** [comment]
Haven't checked the code in detail, but make sure that other scenes are saved only if they are marked as modified.

**Comment by MarianoGnu** [comment]
Well not sure how to check that actually, it looks like they are unsaved when version!=0 is this correct?

File: tools/editor/editor_node.cpp [hunk 2]
```diff
@@ -1881,7 +1878,7 @@ void EditorNode::_run(bool p_current,const String& p_custom) {
 				_save_scene_with_preview(scene->get_filename());
 			}
 		}
-
+		_menu_option(FILE_SAVE_ALL_SCENES);
```
Comment (MarianoGnu) [inline tools/editor/editor_node.cpp]: I should add the version check to this line right?

File: tools/editor/editor_node.cpp [hunk 3]
```diff
@@ -2138,6 +2135,19 @@ void EditorNode::_menu_option_confirm(int p_option,bool p_confirmed) {
 
 		} break;
 
+		case FILE_SAVE_ALL_SCENES: {
+			for (int i = 0; i < editor_data.get_edited_scene_count(); i++) {
+				Node *scene = editor_data.get_edited_scene_root(i);
+				if (scene && scene->get_filename()!="") {
```
Comment (MarianoGnu) [inline tools/editor/editor_node.cpp]: I should add the version check to this line

**Comment by MarianoGnu** [comment]
looks like i forgot to update _set_scene_metadata too, please don't merge yet

**Comment by reduz** [comment]
how is this coming along? there seems to be a chain of PRs and issues linked to this being fixed

**Comment by MarianoGnu** [comment]
Ah, ok i'll take a,look today, there's a missing detail, scenes in background ase not saving plugins's states

**Comment by reduz** [comment]
the plugin state is still contained in the state dictionary I think, just
check EditorNode::save_scene

**Comment by reduz** [comment]
also make sure that scenes that are marked as saved (no changes) are not
saved, otherwise it will take a long time saving if you have many open

**Comment by MarianoGnu** [comment]
I belive this is ready, PLEASE take special care when reviewing this, since saving scenes is a very delicate process -_- i don't trust myself

**Comment by MarianoGnu** [comment]
Something is wrong, looks like the scene states are not loading u.u

**Comment by MarianoGnu** [comment]
all right, i think it's ok now...

**Comment by reduz** [comment]
Seems good to me, though I wish more people could test it before merging it

**Comment by reduz** [comment]
Also once this is merged, #5385 should be merged too

**Pull request merged**